typescript/infra/scripts/check/check-warp-deploy.ts

```diff
@@ -47,17 +55,26 @@ async function main() {
     console.log(chalk.yellow('Skipping the following warp routes:'));
     routesToSkip.forEach((route) => console.log(chalk.yellow(`- ${route}`)));
 
-    warpIdsToCheck = Object.keys(warpConfigGetterMap).filter(
-      (warpRouteId) => !routesToSkip.includes(warpRouteId),
-    );
+    warpIdsToCheck = Object.keys(
+      await getWarpConfigMapFromMergedRegistry(
+        registry ?? [DEFAULT_REGISTRY_URI],
+      ),
+    ).filter((warpRouteId) => !routesToSkip.includes(warpRouteId));
   }
 
   // Determine which chains have warp configs
-  const chainsWithWarpConfigs = warpIdsToCheck.reduce((chains, warpRouteId) => {
-    const warpAddresses = getWarpAddresses(warpRouteId);
-    Object.keys(warpAddresses).forEach((chain) => chains.add(chain));
-    return chains;
-  }, new Set<ChainName>());
+  const chainsWithWarpConfigs = new Set<ChainName>();
+  await Promise.all(
```

Comment:
Updated because getWarpAddressesFromMergedRegistry() is async and reduce doesnt work with async